lib/autorestPlugin/pluginHost.js

```diff
@@ -0,0 +1,204 @@
+            if (request.isValid === false) {
+              const error = request.error;
+              const innerErrors = error.innerErrors;
+              if (!innerErrors || !innerErrors.length) {
+                throw new Error("Model Validator: Unexpected format.");
+              }
+              for (const innerError of innerErrors) {
+                const path = ConvertIndicesFromStringToNumbers(innerError.path);
+                //console.error(JSON.stringify(error, null, 2));
+                result = new FormattedOutput("error", error, [error.code, error.id, "ExampleModelViolation"],
+                  innerError.message + ". \nScenario: " + scenario + ". \nDetails: " + JSON.stringify(innerError.errors, null, 2) + "\nMore info: " + openAPIDocUrl + "#" + error.id.toLowerCase() + "-" + error.code.toLowerCase() + "\n",
+                  [{ document: swaggerFileName, Position: { path: path } }]);
+                formattedResult.push(result);
+              }
+            }
+
+            // responses
+            for (var responseCode in scenarioItem.responses) {
+              const response = scenarioItem.responses[responseCode];
+              if (response.isValid === false) {
+                const error = response.error;
+                const innerErrors = error.innerErrors;
+                if (!innerErrors || !innerErrors.length) {
+                  throw new Error("Model Validator: Unexpected format.");
+                }
+                for (const innerError of innerErrors) {
+                  //console.error(JSON.stringify(error, null, 2));
+                  result = new FormattedOutput("error", error, [error.code, error.id, "ExampleModelViolation"],
+                    innerError.message + ". \nScenario: " + scenario + ". \nDetails: " + JSON.stringify(innerError.errors, null, 2) + "\nMore info: " + openAPIDocUrl + "#" + error.id.toLowerCase() + "-" + error.code.toLowerCase() + "\n",
+                    [{ document: swaggerFileName, Position: { path: xmsexPath.slice(0, xmsexPath.length - 1).concat(["responses", responseCode]) } }
+                    ])
+                  formattedResult.push(result);
+                }
+              }
+            }
+          }
+        }
+      }
+    })
+    return formattedResult;
+  }).catch(function (err) {
+    console.error(err);
+    return Promise.reject(err);
+  });
+};
+/**
+ * Path comes with indices as strings in "inner errors", so converting those to actual numbers for path to work.
+ */
+function ConvertIndicesFromStringToNumbers(path) {
+  const result = path.slice();
+  for (let i = 1; i < result.length; ++i) {
+    const num = parseInt(result[i]);
+    if (!isNaN(num) && result[i - 1] === "parameters") {
+      result[i] = num;
+    }
+  }
+  return result;
+}
+
+class PluginHost {
+  constructor(initiator) {
+    this.initiator = initiator;
+  }
+
+  GetPluginNames() {
+    return [/*OpenApiValidationSemantic.Name, */ OpenApiValidationExample.Name];
```

Comment:
Oh, it's commented out, because I haven't hooked up Semantic validation yet, it's coming...